test/unit/services/proxy_deployment_service_test.rb

```diff
@@ -47,26 +46,6 @@ def setup
     assert service.call
   end
 
-  test 'deploy neither service mesh or apicast configuration driven' do
-    @proxy.expects(:service_mesh_integration?).returns(false)
-    @proxy.expects(:apicast_configuration_driven).returns(false)
-
-    service = ProxyDeploymentService.new(@proxy)
-
-    refute service.call
-  end
```

Comment:
The expectation has changed, `deploy` will now always return even if neither service mesh or apicast config driven. So, instead of removing this test, change the assertion to:
```diff
+service.expects(:deploy_v2).once
+service.expects(:deploy_production_v2).once

- refute service.call
+ assert service.call
```

Reply:
now we are checking only the apicast_configuration_driven? conditionhttps://github.com/3scale/porta/pull/2495#discussion_r660455415

Reply:
Then aren't we executing a conditional branch instead of NOT executing any (like it was in the past). Hery stated above that it should be harmless but the above assessment suggests otherwise.